modules/programs/vscode.nix

```diff
@@ -164,10 +181,16 @@ in {
     home.packages = [ cfg.package ];
 
     home.file = mkMerge [
-      (mkIf (cfg.userSettings != { }) {
-        "${configFilePath}".source =
-          jsonFormat.generate "vscode-user-settings" cfg.userSettings;
-      })
+      (mkIf (cfg.userSettings != { } || cfg.disableUpdateCheck
+        || cfg.disableExtensionUpdateCheck) {
+          "${configFilePath}".source =
+            jsonFormat.generate "vscode-user-settings" (cfg.userSettings
+              // optionalAttrs (cfg.disableUpdateCheck) {
+                "update.mode" = "none";
+              } // optionalAttrs (cfg.disableExtensionUpdateCheck) {
+                "extensions.autoCheckUpdates" = false;
+              });
+        })
```

Comment:
It may be cleaner to merge `cfg.userSettings` and the two optional attr sets and then check if that is `{ }`. Then, just generate the whole config.

Something like:

```nix
let computedUserSettings = cfg.userSettings
  // optionalAttrs (cfg.disableUpdateCheck) {
    # ...
  }
  // optionalAttrs (cfg.disableExtensionUpdateCheck) {
    # ...
  };
in mkIf (computedUserSettings != { }) {
  # ...
}
```

Reply:
I agree.